File: frontend/src/api/equipment.ts
import axios from 'axios'

const API_BASE_URL = '/api/v1'

const apiClient = axios.create({
  baseURL: API_BASE_URL,
  timeout: 300000,
  headers: {
    'Content-Type': 'application/json'
  }
})

apiClient.interceptors.response.use(
  response => response,
  error => {
    console.error('API Error:', error)
    return Promise.reject(error)
  }
)

export interface TableInfo {
  table_name: string
  columns: string[]
  row_count: number
  preview?: Record<string, any>[]
}

export interface AnalysisRecord {
  id: string
  file_name: string
  file_size: number
  file_type: string
  table_count: number
  record_count: number
  status: string
  analysis_result?: any
  error_message?: string
  created_at: string
  completed_at?: string
}

export const equipmentApi = {
  uploadFile: async (file: File) => {
    const formData = new FormData()
    formData.append('file', file)
    const response = await apiClient.post('/upload', formData, {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    })
    return response.data
  },

  getRecords: async (skip = 0, limit = 20) => {
    const response = await apiClient.get('/records', {
      params: { skip, limit }
    })
    return response.data
  },

  getRecord: async (id: string) => {
    const response = await apiClient.get(`/records/${id}`)
    return response.data
  },

  getRecordTables: async (id: string) => {
    const response = await apiClient.get(`/records/${id}/tables`)
    return response.data
  },

  getTableData: async (recordId: string, tableName: string, page = 1, pageSize = 100) => {
    const response = await apiClient.get(`/records/${recordId}/tables/${tableName}`, {
      params: { page, page_size: pageSize }
    })
    return response.data
  },

  analyzeData: async (recordId: string, query?: string, useLocalModel = false) => {
    const response = await apiClient.post('/analyze', {
      record_id: recordId,
      query,
      use_local_model: useLocalModel
    })
    return response.data
  },

  deleteRecord: async (id: string) => {
    const response = await apiClient.delete(`/records/${id}`)
    return response.data
  },

  healthCheck: async () => {
    const response = await apiClient.get('/health')
    return response.data
  }
}

export default apiClient

File: backend/app/schemas/equipment.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime

class AnalysisRecordBase(BaseModel):
    file_name: str
    file_size: int
    file_type: str
    table_count: int = 0
    record_count: int = 0
    status: str = "pending"

class AnalysisRecordCreate(AnalysisRecordBase):
    pass

class AnalysisRecordResponse(AnalysisRecordBase):
    id: str
    table_name: Optional[str] = None
    source_record_id: Optional[str] = None
    analysis_result: Optional[Dict[str, Any]] = None
    error_message: Optional[str] = None
    created_at: datetime
    completed_at: Optional[datetime] = None

    class Config:
        from_attributes = True

class TableInfo(BaseModel):
    table_name: str
    columns: List[str]
    row_count: int
    preview: List[Dict[str, Any]]

class FileParseResult(BaseModel):
    file_name: str
    file_size: int
    file_type: str
    tables: List[TableInfo]
    total_records: int
    parsed_at: str

class AnalyzeRequest(BaseModel):
    record_id: str
    table_name: Optional[str] = None
    query: Optional[str] = None
    use_local_model: bool = False

class AnalyzeResponse(BaseModel):
    record_id: str
    status: str
    result: Optional[Dict[str, Any]] = None
    message: str
